src/Oro/Bundle/FilterBundle/Filter/NumberFilter.php

```diff
@@ -81,9 +81,43 @@ public function getMetadata()
     {
         $metadata = parent::getMetadata();
 
-        $formView = $this->getForm()->createView();
-        $metadata['formatterOptions'] = $formView->vars['formatter_options'];
+        $formBuilder = $this->getFormBuilder();
+        $dataType = $formBuilder->getOption('data_type');
+        $metadata['formatterOptions'] = $this->getFormatterOptions($dataType);
 
         return $metadata;
     }
+
+    /**
+     * TODO: Move this in another service?
+     *
+     * @param string $dataType
+     *
+     * @return array
+     */
+    public function getFormatterOptions($dataType)
```

Comment:
This method is a copy paste from the logic in the `NumberFilterType` here:
https://github.com/akeneo/platform/blob/d126ee653b703dc199b4a6c119a1cef6866192d5/src/Oro/Bundle/FilterBundle/Form/Type/Filter/NumberFilterType.php#L82-L106